lib/matplotlib/axis.py

```diff
@@ -446,15 +446,8 @@ def _get_gridline(self):
 
         return l
 
-    def update_position(self, loc):
-        'Set the location of tick in data coords with scalar *loc*'
-        x = loc
-
-        nonlinear = (hasattr(self.axes, 'yaxis') and
-                     self.axes.yaxis.get_scale() != 'linear' or
-                     hasattr(self.axes, 'xaxis') and
-                     self.axes.xaxis.get_scale() != 'linear')
-
+    def update_position(self, x):
```

Comment:
Please do not change the arg names, it will break anyone who is unpacking dictionaries in to this method.